src/config/job-config/actions/rabbitmqaction/rabbitmqaction.ts

```
@@ -1,91 +1,43 @@
 import { Logger } from "@nestjs/common";
-import amqp, { Connection } from "amqplib/callback_api";
 import { JobAction, JobDto } from "../../jobconfig.interface";
 import { JobClass } from "../../../../jobs/schemas/job.schema";
 import {
   actionType,
   RabbitMQJobActionOptions,
 } from "./rabbitmqaction.interface";
+import { RabbitMQService } from "src/common/rabbitmq/rabbitmq.service";
 
 /**
  * Publish a message in a RabbitMQ queue
  */
 export class RabbitMQJobAction<T extends JobDto> implements JobAction<T> {
-  public static readonly actionType = "rabbitmq";
-  private connection;
-  private binding;
+  private queue: string;
+  private exchange: string;
+  private key: string;
 
-  constructor(options: RabbitMQJobActionOptions) {
+  getActionType(): string {
+    return actionType;
+  }
+
+  constructor(
+    private readonly rabbitMQService: RabbitMQService,
+    options: RabbitMQJobActionOptions,
+  ) {
     Logger.log(
       "Initializing RabbitMQJobAction. Params: " + JSON.stringify(options),
       "RabbitMQJobAction",
     );
-
-    this.connection = {
-      protocol: "amqp",
-      hostname: options.hostname,
-      port: options.port,
-      username: options.username,
-      password: options.password,
-    };
-    this.binding = {
-      exchange: options.exchange,
-      queue: options.queue,
-      key: options.key,
-    };
-  }
-
-  getActionType(): string {
-    return actionType;
+    this.queue = options.queue;
```

should throw an error if rabbitMQ isn't enabled